utest/src-0/utest/Errors.scala

```diff
@@ -0,0 +1,98 @@
+package utest
```

Comment:
Is there a reason that this file cannot be shared between Scala 2 and Dotty? Seems like mostly plain-old-Scala code that doesn't do anything fancy

Reply:
We can probably share it. Only the error reporting logic was changed for Dotty, as I wrote earlier:

> Errors.scala which checks compile-time errors has flaky reporting. It looks like the indentation of the error messages conflicts with the general indentation of the test output (the latter is indented by two spaces). If you print the detected vs expected error messages from the assert output in a standard utest way, the position string of the compiler error message will get messed up (wrong indentation) under certain conditions. Therefore I changed the error reporting for Dotty to use println for compiler error messages. Ugly, but precise.

I haven't investigated whether the above flaky behaviour is also present in Scala 2. If it is, I would advocate for moving the `println` way of doing things until a better solution is found; if it isn't, we can split only the `check` method between the two Scala versions and share the rest.

Reply:
If you can't be bothered to fix it for now, splitting out the `check` method would be enough for now (with a comment on the Dotty version with the quoted text above, explaining why it needs to be different)

Reply:
Doesn't reproduce anymore. I've shared the logic between Scala 2 and Scala 3 largely.